### ISSUE TITLE
Support bi-directional patching in `crossplane beta render`

### ISSUE THREAD

**Issue by ytsarev** [crossplane/crossplane#5196]
### What problem are you facing?

As was initially reported in https://github.com/crossplane/crossplane/issues/5124#issuecomment-1853570772

`crossplane beta render` does not support any form of bi-directional patching, meanwhile it is a typical pattern to share the data between composed resources in Patch and Transform as documented in https://docs.crossplane.io/latest/concepts/patch-and-transform/#patching-between-resources.

There are also occasions when people are pushing to status while implementing Composition Function and want to reuse the function-generated value from XR status in another consequent Pipeline step.

### How could Crossplane help solve your problem?

Support bi-directional patching scenario with `crossplane beta render`

**Comment by phisco** [comment]
Patching between resources depends on the function, what you really want is `crossplane beta render` to support multiple iterations, but I think it would be out of scope for the render subcommand, but rather should be something we build on top of that, a sort of testing framework where you can define at each iterations inputs, expected outputs and wether you want to feed such output to another iteration. This could be done today with some bash scripting, but it would be definitely better to have a separate subcommand doing that with some sort of configuration file describing your scenarios and some convention for the manifests, similar to KUTTL.

**Comment by ytsarev** [comment]
@phisco in this case I would like to have reliable automatically generated inputs, ideally produced by `crossplane beta render` so it might sense to incorporate it to the `beta render` tooling

**Comment by phisco** [comment]
@ytsarev do you mean like being able to directly apply the desired state on top of the previously observed resources?
Something like the diagram below?

![image](https://github.com/crossplane/crossplane/assets/5697904/2ae2ece3-f892-4b42-9abb-ad1265c4006c)

**Comment by ytsarev** [comment]
Exactly! so the output of the first t0 iteration can act as input to Observed resource map for the second t1.

**Comment by phisco** [comment]
Got it! I agree it would be a useful feature, but I think there is value in keeping `crossplane beta render` as simple as possible, it should be a tool to help developing a single composition, trying to fit in there also a full testing framework would be a bit of a stretch. Something else can definitely be built reusing the same engine, either directly in the crossplane cli and importing it directly, or maybe even just reusing the crossplane cli at first.

Assuming you had a magic command, `crossplane alpha test`, how would that work? How would you like to define your steps?

**Comment by excalq** [comment]
Perhaps rather than failing to render a resource with an error (as shown below), `crossplane beta render` could instead treat `status` values as Terraform does with "`(known after apply)`" values, still rendering the rest of the XR for previewing.

```
apiVersion: render.crossplane.io/v1beta1
kind: Result
message: 'cannot render FromComposite patches for composed resource "xReplicationGroup":
  cannot apply the "FromCompositeFieldPath" patch at index 7: status: no such field'
severity: SEVERITY_WARNING
step: patch-and-transform
```

**Comment by kaessert** [comment]
Leaving my thoughts here from a previous conversation:
```
What would be nice to me would be the ability to specify input-manifests as patch-sets instead of the full state and maybe even the option to run all in a row :thinking_face: for example
let's say i have a folder (render-inputs for example)
there i have files (test-happy1.yaml, test-happy2.yaml, test-fail.yaml)
test-files contain something like
kubernetesCluster:
  status:
    atProvider:
      some: field

xr:
  ...
now idea was to have `crossplane render --patches="test-happy1.yaml,test-happy-2.yaml" doing:
run normal render with empty input
grab the output from render, apply the first patch and use that as an oberved-input for the next render which will serve as an input for the next patch
rinse and repeat for all files
I hope it makes sense, it's just from the top of my head. Would solve two problems for me:
less verbose method of specifying input-data
better ability to chain renders in the same way as it would progress in the real world
```
